### ISSUE TITLE
[NamedPipeServer][Error  ] Pipe failure: Pipe hasn't been connected yet

### ISSUE THREAD

**Issue by yvesdelvaux** [docker/for-win#482]
### Expected behavior
I can restart Docker without restart my pc.

### Actual behavior
Docker doesn't restart and freeze.

### Information

  - Diagnostic ID : 6C2349F8-BC33-4B0F-9D88-DA5E76C8BD03/2017-02-10_14-32-36
  - Reproductible case : Update the docker or restart it
  - host distribution and version (Windows version, build number, etc) Windows 10 14393

### Steps to reproduce the behavior

  1. Then the Docker is running, change anything in configuration and restart it
  2. Docker is in the state "Docker is starting" but it doesn't start. It seems to be in freeze state.

**Comment by rneugeba** [comment]
Thanks for your report. Something seems to have gone wrong during the upgrade process. It's not clear what, but could you try:
- Reset to factory defaults (from the whale menu, Settings, reset)
- If that doesn't work, manually uninstall docker and download the latest installer from https://docs.docker.com/docker-for-windows/

Thanks

**Comment by rneugeba** [comment]
@mmisztal1980 reported the same error with exactly the same error message on https://github.com/docker/for-win/issues/487 with diagnostics id 5239E3DB-D0D2-47AE-B158-5D03C5A0619A/2017-02-10_22-19-15

the error message is:
```
[16:35:11.136][NamedPipeServer][Error  ] Pipe failure: Pipe hasn't been connected yet.    at System.IO.Pipes.PipeStream.CheckReadOperations()
   at System.IO.Pipes.PipeStream.BeginRead(Byte[] buffer, Int32 offset, Int32 count, AsyncCallback callback, Object state)
   at System.IO.Stream.<>c.<BeginEndReadAsync>b__43_0(Stream stream, ReadWriteParameters args, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncTrim[TInstance,TArgs](TInstance thisRef, TArgs args, Func`5 beginMethod, Func`3 endMethod)
   at System.IO.Stream.BeginEndReadAsync(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.Stream.ReadAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at System.IO.Stream.ReadAsync(Byte[] buffer, Int32 offset, Int32 count)
   at Docker.Core.Pipe.NamedPipeServer.<HandleRequestAsync>d__11.MoveNext()
[16:35:11.156][Notifications  ][Error  ] Binary stream '0' does not contain a valid BinaryHeader. Possible causes are invalid stream or object version change between serialization and deserialization.
```

**Comment by dvprao** [comment]
https://github.com/docker/for-win/issues
CAFC569A-DA17-4E8C-ABA5-D9AF5DFA1809/2017-02-10_16-08-08
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.26/containers/json: open //./pipe/docker_engine: The system cannot find the file specified. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.

Very frustrating for a newbie.

**Comment by rneugeba** [comment]
@dvprao thanks for the report. I have already filed on internal issue

**Comment by mmisztal1980** [comment]
@rneugeba It's impossible to use the "reset". All the options in the "Reset" pane are grayed out.

**Comment by mmisztal1980** [comment]
@rneugeba I've reinstalled docker for windows using the latest **stable** installer. Same issue.

**Comment by rneugeba** [comment]
@mmisztal1980 could you provide a diagnostics ID? thanks

**Comment by mmisztal1980** [comment]
Latest uploaded diagnostic: 5239E3DB-D0D2-47AE-B158-5D03C5A0619A/2017-02-11_10-55-40

**Comment by rneugeba** [comment]
@mmisztal1980 thanks for the quick response. Your issue seems different. could you file a separate issue? Thanks

**Comment by mmisztal1980** [comment]
I already did, but you closed it :D

**Comment by rneugeba** [comment]
Ah, sorry. Yes, you other logs under 5239E3DB-D0D2-47AE-B158-5D03C5A0619A/2017-02-10_22-19-15 do show this error. Sorry, for the confusion

**Comment by mmisztal1980** [comment]
No problem, I suppose we can reopen it and continue there?

**Comment by rneugeba** [comment]
no, your other logs contain the same error as the report on this issue. we need to look into why there is a backtrace and why subsequently nothing works anymore.

Did older versions work for you?
Did you install/update any software, in particular AV software?

**Comment by mmisztal1980** [comment]
It used to work up to a week ago, there was an update recently, but it didn't work by then. 
My AV software is ESET NOD32, but I haven't made any updates to it's configuration recently.

**Comment by rneugeba** [comment]
Could you try disabling the AV software and see if it works?

Also, could you try to revert to 1.12.6 from here https://download.docker.com/win/stable/1.12.6.9655/InstallDocker.msi

and also 1.13.0 from here:  https://download.docker.com/win/stable/1.13.0.9795/InstallDocker.msi

that would help us narrow down if it's something we changed.

Thanks

**Comment by mmisztal1980** [comment]
I've disabled the following features in NOD:
- HIPS (host intrusion prevention system)
- Real-time file system protection

No effect, docker is still "stuck" in "Docker is starting state". I'll try the older versions next.

**Comment by mmisztal1980** [comment]
##v1.12.6 :

Docker seems to be stuck again. Last meaninfgul enrtry in the log suggests that it was trying to create the Hyper-V NAT switch. 

```Version: 1.12.6 (9655)
Channel: Stable
Sha1: 258480fd147da7888a440ed7808d255c0fb15bb7
Started on: 2017/02/11 13:05:41.563
Resources: C:\Program Files\Docker\Docker\Resources
OS: Windows 10 Enterprise
Edition: Enterprise
Id: 1607
Build: 14393
BuildLabName: 14393.693.amd64fre.rs1_release.161220-1747
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[13:05:41.509][Tracking       ][Info   ] Crash report and usage statistics are enabled
[13:05:41.585][SegmentApi     ][Info   ] Usage statistic: appLaunched
[13:05:41.653][NamedPipeClient][Info   ] Sending Version()...
[13:05:41.653][NamedPipeClient][Info   ] Received response for Version
[13:05:41.669][SegmentApi     ][Info   ] Usage statistic: heartbeat
[13:05:36.255][BackendServer  ][Info   ] Started
[13:05:36.256][Moby           ][Info   ] Connecting...
[13:05:41.653][NamedPipeServer][Info   ] Version()
[13:05:41.653][NamedPipeServer][Info   ] Version done.
[13:05:41.872][Notifications  ][Info   ] Docker is starting...
[13:05:41.872][Updater        ][Info   ] Checking for updates on channel Stable...
[13:05:41.887][NamedPipeClient][Info   ] Sending Start(Docker.Core.Settings)...
[13:05:41.887][NamedPipeServer][Info   ] Start(Docker.Core.Settings)
[13:05:41.887][HyperVGuids    ][Info   ] Installing GUIDs...
[13:05:41.903][HyperVGuids    ][Info   ] GUIDs installed
[13:05:41.903][Firewall       ][Info   ] Opening ports for C:\Program Files\Docker\Docker\Resources\com.docker.proxy.exe...
[13:05:42.012][Updater        ][Info   ] We got a new version 10072 which is newer than 9655, asking user.
[13:05:42.012][SegmentApi     ][Info   ] Usage statistic: eventShowUpdateWindows
[13:05:42.327][Updater        ][Info   ] Check for updates done.
[13:05:42.984][Firewall       ][Info   ] Ports are opened
[13:05:42.986][HyperV         ][Info   ] Create
[13:05:42.993][PowerShell     ][Info   ] Run script with parameters: -Create True -SwitchSubnetAddress 10.0.75.0 -SwitchSubnetMaskSize 24 -CPUs 2 -Memory 2048 -IsoFile C:\Program Files\Docker\Docker\Resources\mobylinux.iso...
[13:05:45.144][HyperV         ][Info   ] Creating Switch: DockerNAT...
[13:06:14.242][SegmentApi     ][Info   ] Usage statistic: actionMenuSettings
[13:06:18.475][FeedbackSettings][Info   ] Open logs
[13:08:02.903][SegmentApi     ][Info   ] Usage statistic: actionMenuSettings
[13:09:17.835][FeedbackSettings][Info   ] Open logs
[13:10:24.675][SegmentApi     ][Info   ] Usage statistic: actionMenuFeedback
[13:10:26.063][FeedbackSettings][Info   ] Upload diagnostic
[13:10:26.066][NamedPipeClient][Info   ] Sending GetDebugInfo()...
[13:10:26.068][NamedPipeServer][Info   ] GetDebugInfo()
[13:10:37.591][NamedPipeServer][Info   ] GetDebugInfo done.
[13:10:37.605][NamedPipeClient][Info   ] Received response for GetDebugInfo
[13:10:37.612][Program        ][Info   ] Sending Bugsnag report e2ff260c-062c-4251-9040-c0ce6d2a7fb8...
[13:10:39.181][Program        ][Info   ] Bugsnag report e2ff260c-062c-4251-9040-c0ce6d2a7fb8 sent
[13:10:39.200][NamedPipeClient][Info   ] Sending DownloadVmLogs()...
[13:10:39.201][NamedPipeServer][Info   ] DownloadVmLogs()
[13:10:39.204][PowerShell     ][Info   ] Run script...
```

I'm attempting to upload a diagnostic but I'm not getting the UI to present me with the diagnostic's ID.

**Comment by yvesdelvaux** [comment]
Thanks for the answer. I have installed the new version manually. That's work for the moment. But the next update can fail if this issue is not solved.

I already have this problem at the previous update. The issue appears with the Windows container version of docker.

**Comment by cyberjoac** [comment]
+1

**Comment by rneugeba** [comment]
@yvesdelvaux thanks for getting back. This is weird...Do you have any AV software installed?

@cyberjoac can you provide a diagnostic ID?

**Comment by cyberjoac** [comment]
Just uploaded the diagnostic ID: 639B87D4-CE3B-4850-90DC-A38C83D12E6A/2017-02-12_16-01-57

Please note I can't see any VM in the Hyper-V manager..

**Comment by mmisztal1980** [comment]
@rneugeba while using v12, I was consistently unable to upload any diagnostics.
I've updated to v13.0. The main issue persists, however I'm now able to upload diagnostics again, so :
I've uploaded a new diagnostic : 5239E3DB-D0D2-47AE-B158-5D03C5A0619A/2017-02-12_17-37-41

**Comment by mmisztal1980** [comment]
Another observation:
v12: while navigating the panes of the docker for windows application I'm aboe to see controls' contents, such as Subnet Mask, Subnet Address etc. - can't see any content in v13.

**Comment by yvesdelvaux** [comment]
@rneugeba Yes I have an AV (Trend Micro OfficeScan). I can't disable it. My firm has a strict security policy about it.

**Comment by rneugeba** [comment]
@cyberjoac thanks for the diagnostics ID. You indeed have the same issue as others.

@mmisztal1980 hmm, odd. Could you attach a screen shot from the 1.13 control panel? Thanks

**Comment by mmisztal1980** [comment]
@rneugeba here you go:
![docker-v13-issues](https://cloud.githubusercontent.com/assets/1357057/22880999/c3f9a48e-f1e4-11e6-88fa-729049f9cf14.png)

**Comment by rneugeba** [comment]
It's not clear to me where the issue is, so, it would be good if you could try running  a bunch of experiments so we could try to narrow down the area.

Before doing these, please make sure you are on the latest beta or stable version. Then quit the application and stop the `com.docker.service` by issuing `Stop-Service com.docker.service` from and elevated powershell.  Also clear all the log files from `~\AppData\Local\Docker\`.

## UI test
With the application exited and the service quit, with an elevated powershell go to `C:\Program Files\Docker\Docker\resources` and issue `.\MobyLinux.ps1 -destroy`. In Hyper-V Manager (or via `Get-VM`) make sure the `MobyLinuxVM` VM and `DockerNAT` switch is removed

Now start the application. You should get prompted to start the service and then with a UAC prompt.

We'd be interested in the logs/diagnostics if this fails.

## Hyper-V test
With an elevated powershell go to `C:\Program Files\Docker\Docker\resources`:
- `.\MobyLinux.ps1 -destroy`. In Hyper-V Manager (or via `Get-VM`) make sure the `MobyLinuxVM` VM and `DockerNAT` switch is removed . 
- `.\MobyLinux.ps1 -create`. In Hyper-V Manager (or via `Get-VM` and `Get-VMSwitvh`) make sure the `MobyLinuxVM` VM and the `DockerNAT` switch is created.
- `.\MobyLinux.ps1 -start`. In Hyper-V Manager (or via `Get-VM`) make sure the state of the `MobyLinuxVM` changes to running.

If all the above is working without error, the issue is unlikely around Hyper-V...Execute `.\MobyLinux.ps1 -stop` followed by `.\MobyLinux.ps1 -destroy` to clean up.

Thanks for your help

**Comment by mmisztal1980** [comment]
@rneugeba proceeding with your suggestions, please give me 1-2hrs to complete, as I'm at work right now :)

**Comment by mmisztal1980** [comment]
@rneugeba 

#UI Test:

Diagnostic uploaded: 5239E3DB-D0D2-47AE-B158-5D03C5A0619A/2017-02-13_13-19-31

Attaching all logs since the install of the latest beta. After install, I've shut it down to ensure that the MobyLinuxVM is destroyed as you've requested.

[DebugInfo-2017-02-13_13-19-31.txt](https://github.com/docker/for-win/files/771001/DebugInfo-2017-02-13_13-19-31.txt)
[install-log.txt](https://github.com/docker/for-win/files/771000/install-log.txt)
[log.0.txt](https://github.com/docker/for-win/files/770998/log.0.txt)
[log.txt](https://github.com/docker/for-win/files/770999/log.txt)

I'll proceed with the Hyper-V test in 10 minutes.

**Comment by mmisztal1980** [comment]
# Hyper-V Test

```
PS C:\Program Files\Docker\Docker\resources> .\MobyLinux.ps1 -create -verbose
Script started at 14:16:32.345
VERBOSE: Importing cmdlet 'Add-Member'.
VERBOSE: Importing cmdlet 'Add-Type'.
VERBOSE: Importing cmdlet 'Clear-Variable'.
VERBOSE: Importing cmdlet 'Compare-Object'.
VERBOSE: Importing cmdlet 'Convert-String'.
VERBOSE: Importing cmdlet 'ConvertFrom-Csv'.
VERBOSE: Importing cmdlet 'ConvertFrom-Json'.
VERBOSE: Importing cmdlet 'ConvertFrom-String'.
VERBOSE: Importing cmdlet 'ConvertFrom-StringData'.
VERBOSE: Importing cmdlet 'ConvertTo-Csv'.
VERBOSE: Importing cmdlet 'ConvertTo-Html'.
VERBOSE: Importing cmdlet 'ConvertTo-Json'.
VERBOSE: Importing cmdlet 'ConvertTo-Xml'.
VERBOSE: Importing cmdlet 'Debug-Runspace'.
VERBOSE: Importing cmdlet 'Disable-PSBreakpoint'.
VERBOSE: Importing cmdlet 'Disable-RunspaceDebug'.
VERBOSE: Importing cmdlet 'Enable-PSBreakpoint'.
VERBOSE: Importing cmdlet 'Enable-RunspaceDebug'.
VERBOSE: Importing cmdlet 'Export-Alias'.
VERBOSE: Importing cmdlet 'Export-Clixml'.
VERBOSE: Importing cmdlet 'Export-Csv'.
VERBOSE: Importing cmdlet 'Export-FormatData'.
VERBOSE: Importing cmdlet 'Export-PSSession'.
VERBOSE: Importing cmdlet 'Format-Custom'.
VERBOSE: Importing cmdlet 'Format-List'.
VERBOSE: Importing cmdlet 'Format-Table'.
VERBOSE: Importing cmdlet 'Format-Wide'.
VERBOSE: Importing cmdlet 'Get-Alias'.
VERBOSE: Importing cmdlet 'Get-Culture'.
VERBOSE: Importing cmdlet 'Get-Date'.
VERBOSE: Importing cmdlet 'Get-Event'.
VERBOSE: Importing cmdlet 'Get-EventSubscriber'.
VERBOSE: Importing cmdlet 'Get-FormatData'.
VERBOSE: Importing cmdlet 'Get-Host'.
VERBOSE: Importing cmdlet 'Get-Member'.
VERBOSE: Importing cmdlet 'Get-PSBreakpoint'.
VERBOSE: Importing cmdlet 'Get-PSCallStack'.
VERBOSE: Importing cmdlet 'Get-Random'.
VERBOSE: Importing cmdlet 'Get-Runspace'.
VERBOSE: Importing cmdlet 'Get-RunspaceDebug'.
VERBOSE: Importing cmdlet 'Get-TraceSource'.
VERBOSE: Importing cmdlet 'Get-TypeData'.
VERBOSE: Importing cmdlet 'Get-UICulture'.
VERBOSE: Importing cmdlet 'Get-Unique'.
VERBOSE: Importing cmdlet 'Get-Variable'.
VERBOSE: Importing cmdlet 'Group-Object'.
VERBOSE: Importing cmdlet 'Import-Alias'.
VERBOSE: Importing cmdlet 'Import-Clixml'.
VERBOSE: Importing cmdlet 'Import-Csv'.
VERBOSE: Importing cmdlet 'Import-LocalizedData'.
VERBOSE: Importing cmdlet 'Import-PSSession'.
VERBOSE: Importing cmdlet 'Invoke-Expression'.
VERBOSE: Importing cmdlet 'Invoke-RestMethod'.
VERBOSE: Importing cmdlet 'Invoke-WebRequest'.
VERBOSE: Importing cmdlet 'Measure-Command'.
VERBOSE: Importing cmdlet 'Measure-Object'.
VERBOSE: Importing cmdlet 'New-Alias'.
VERBOSE: Importing cmdlet 'New-Event'.
VERBOSE: Importing cmdlet 'New-Object'.
VERBOSE: Importing cmdlet 'New-TimeSpan'.
VERBOSE: Importing cmdlet 'New-Variable'.
VERBOSE: Importing cmdlet 'Out-File'.
VERBOSE: Importing cmdlet 'Out-GridView'.
VERBOSE: Importing cmdlet 'Out-Printer'.
VERBOSE: Importing cmdlet 'Out-String'.
VERBOSE: Importing cmdlet 'Read-Host'.
VERBOSE: Importing cmdlet 'Register-EngineEvent'.
VERBOSE: Importing cmdlet 'Register-ObjectEvent'.
VERBOSE: Importing cmdlet 'Remove-Event'.
VERBOSE: Importing cmdlet 'Remove-PSBreakpoint'.
VERBOSE: Importing cmdlet 'Remove-TypeData'.
VERBOSE: Importing cmdlet 'Remove-Variable'.
VERBOSE: Importing cmdlet 'Select-Object'.
VERBOSE: Importing cmdlet 'Select-String'.
VERBOSE: Importing cmdlet 'Select-Xml'.
VERBOSE: Importing cmdlet 'Send-MailMessage'.
VERBOSE: Importing cmdlet 'Set-Alias'.
VERBOSE: Importing cmdlet 'Set-Date'.
VERBOSE: Importing cmdlet 'Set-PSBreakpoint'.
VERBOSE: Importing cmdlet 'Set-TraceSource'.
VERBOSE: Importing cmdlet 'Set-Variable'.
VERBOSE: Importing cmdlet 'Show-Command'.
VERBOSE: Importing cmdlet 'Sort-Object'.
VERBOSE: Importing cmdlet 'Start-Sleep'.
VERBOSE: Importing cmdlet 'Tee-Object'.
VERBOSE: Importing cmdlet 'Trace-Command'.
VERBOSE: Importing cmdlet 'Unblock-File'.
VERBOSE: Importing cmdlet 'Unregister-Event'.
VERBOSE: Importing cmdlet 'Update-FormatData'.
VERBOSE: Importing cmdlet 'Update-List'.
VERBOSE: Importing cmdlet 'Update-TypeData'.
VERBOSE: Importing cmdlet 'Wait-Debugger'.
VERBOSE: Importing cmdlet 'Wait-Event'.
VERBOSE: Importing cmdlet 'Write-Debug'.
VERBOSE: Importing cmdlet 'Write-Error'.
VERBOSE: Importing cmdlet 'Write-Host'.
VERBOSE: Importing cmdlet 'Write-Information'.
VERBOSE: Importing cmdlet 'Write-Output'.
VERBOSE: Importing cmdlet 'Write-Progress'.
VERBOSE: Importing cmdlet 'Write-Verbose'.
VERBOSE: Importing cmdlet 'Write-Warning'.
VERBOSE: Importing function 'ConvertFrom-SddlString'.
VERBOSE: Importing function 'Format-Hex'.
VERBOSE: Importing function 'Get-FileHash'.
VERBOSE: Importing function 'Import-PowerShellDataFile'.
VERBOSE: Importing function 'New-Guid'.
VERBOSE: Importing function 'New-TemporaryFile'.
VERBOSE: Importing alias 'CFS'.
VERBOSE: Importing alias 'fhx'.
VERBOSE: Importing cmdlet 'Add-Computer'.
VERBOSE: Importing cmdlet 'Add-Content'.
VERBOSE: Importing cmdlet 'Checkpoint-Computer'.
VERBOSE: Importing cmdlet 'Clear-Content'.
VERBOSE: Importing cmdlet 'Clear-EventLog'.
VERBOSE: Importing cmdlet 'Clear-Item'.
VERBOSE: Importing cmdlet 'Clear-ItemProperty'.
VERBOSE: Importing cmdlet 'Clear-RecycleBin'.
VERBOSE: Importing cmdlet 'Complete-Transaction'.
VERBOSE: Importing cmdlet 'Convert-Path'.
VERBOSE: Importing cmdlet 'Copy-Item'.
VERBOSE: Importing cmdlet 'Copy-ItemProperty'.
VERBOSE: Importing cmdlet 'Debug-Process'.
VERBOSE: Importing cmdlet 'Disable-ComputerRestore'.
VERBOSE: Importing cmdlet 'Enable-ComputerRestore'.
VERBOSE: Importing cmdlet 'Get-ChildItem'.
VERBOSE: Importing cmdlet 'Get-Clipboard'.
VERBOSE: Importing cmdlet 'Get-ComputerInfo'.
VERBOSE: Importing cmdlet 'Get-ComputerRestorePoint'.
VERBOSE: Importing cmdlet 'Get-Content'.
VERBOSE: Importing cmdlet 'Get-ControlPanelItem'.
VERBOSE: Importing cmdlet 'Get-EventLog'.
VERBOSE: Importing cmdlet 'Get-HotFix'.
VERBOSE: Importing cmdlet 'Get-Item'.
VERBOSE: Importing cmdlet 'Get-ItemProperty'.
VERBOSE: Importing cmdlet 'Get-ItemPropertyValue'.
VERBOSE: Importing cmdlet 'Get-Location'.
VERBOSE: Importing cmdlet 'Get-Process'.
VERBOSE: Importing cmdlet 'Get-PSDrive'.
VERBOSE: Importing cmdlet 'Get-PSProvider'.
VERBOSE: Importing cmdlet 'Get-Service'.
VERBOSE: Importing cmdlet 'Get-TimeZone'.
VERBOSE: Importing cmdlet 'Get-Transaction'.
VERBOSE: Importing cmdlet 'Get-WmiObject'.
VERBOSE: Importing cmdlet 'Invoke-Item'.
VERBOSE: Importing cmdlet 'Invoke-WmiMethod'.
VERBOSE: Importing cmdlet 'Join-Path'.
VERBOSE: Importing cmdlet 'Limit-EventLog'.
VERBOSE: Importing cmdlet 'Move-Item'.
VERBOSE: Importing cmdlet 'Move-ItemProperty'.
VERBOSE: Importing cmdlet 'New-EventLog'.
VERBOSE: Importing cmdlet 'New-Item'.
VERBOSE: Importing cmdlet 'New-ItemProperty'.
VERBOSE: Importing cmdlet 'New-PSDrive'.
VERBOSE: Importing cmdlet 'New-Service'.
VERBOSE: Importing cmdlet 'New-WebServiceProxy'.
VERBOSE: Importing cmdlet 'Pop-Location'.
VERBOSE: Importing cmdlet 'Push-Location'.
VERBOSE: Importing cmdlet 'Register-WmiEvent'.
VERBOSE: Importing cmdlet 'Remove-Computer'.
VERBOSE: Importing cmdlet 'Remove-EventLog'.
VERBOSE: Importing cmdlet 'Remove-Item'.
VERBOSE: Importing cmdlet 'Remove-ItemProperty'.
VERBOSE: Importing cmdlet 'Remove-PSDrive'.
VERBOSE: Importing cmdlet 'Remove-WmiObject'.
VERBOSE: Importing cmdlet 'Rename-Computer'.
VERBOSE: Importing cmdlet 'Rename-Item'.
VERBOSE: Importing cmdlet 'Rename-ItemProperty'.
VERBOSE: Importing cmdlet 'Reset-ComputerMachinePassword'.
VERBOSE: Importing cmdlet 'Resolve-Path'.
VERBOSE: Importing cmdlet 'Restart-Computer'.
VERBOSE: Importing cmdlet 'Restart-Service'.
VERBOSE: Importing cmdlet 'Restore-Computer'.
VERBOSE: Importing cmdlet 'Resume-Service'.
VERBOSE: Importing cmdlet 'Set-Clipboard'.
VERBOSE: Importing cmdlet 'Set-Content'.
VERBOSE: Importing cmdlet 'Set-Item'.
VERBOSE: Importing cmdlet 'Set-ItemProperty'.
VERBOSE: Importing cmdlet 'Set-Location'.
VERBOSE: Importing cmdlet 'Set-Service'.
VERBOSE: Importing cmdlet 'Set-TimeZone'.
VERBOSE: Importing cmdlet 'Set-WmiInstance'.
VERBOSE: Importing cmdlet 'Show-ControlPanelItem'.
VERBOSE: Importing cmdlet 'Show-EventLog'.
VERBOSE: Importing cmdlet 'Split-Path'.
VERBOSE: Importing cmdlet 'Start-Process'.
VERBOSE: Importing cmdlet 'Start-Service'.
VERBOSE: Importing cmdlet 'Start-Transaction'.
VERBOSE: Importing cmdlet 'Stop-Computer'.
VERBOSE: Importing cmdlet 'Stop-Process'.
VERBOSE: Importing cmdlet 'Stop-Service'.
VERBOSE: Importing cmdlet 'Suspend-Service'.
VERBOSE: Importing cmdlet 'Test-ComputerSecureChannel'.
VERBOSE: Importing cmdlet 'Test-Connection'.
VERBOSE: Importing cmdlet 'Test-Path'.
VERBOSE: Importing cmdlet 'Undo-Transaction'.
VERBOSE: Importing cmdlet 'Use-Transaction'.
VERBOSE: Importing cmdlet 'Wait-Process'.
VERBOSE: Importing cmdlet 'Write-EventLog'.
VERBOSE: Importing alias 'gcb'.
VERBOSE: Importing alias 'gin'.
VERBOSE: Importing alias 'gtz'.
VERBOSE: Importing alias 'scb'.
VERBOSE: Importing alias 'stz'.
VERBOSE: Loading module from path 'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\Hyper-V\2.0.0.0\Hyper-V.psd1'.
VERBOSE: Importing cmdlet 'Add-VMAssignableDevice'.
VERBOSE: Importing cmdlet 'Add-VMDvdDrive'.
VERBOSE: Importing cmdlet 'Add-VMFibreChannelHba'.
VERBOSE: Importing cmdlet 'Add-VMGpuPartitionAdapter'.
VERBOSE: Importing cmdlet 'Add-VMGroupMember'.
VERBOSE: Importing cmdlet 'Add-VMHardDiskDrive'.
VERBOSE: Importing cmdlet 'Add-VMHostAssignableDevice'.
VERBOSE: Importing cmdlet 'Add-VMKeyStorageDrive'.
VERBOSE: Importing cmdlet 'Add-VMMigrationNetwork'.
VERBOSE: Importing cmdlet 'Add-VMNetworkAdapter'.
VERBOSE: Importing cmdlet 'Add-VMNetworkAdapterAcl'.
VERBOSE: Importing cmdlet 'Add-VMNetworkAdapterExtendedAcl'.
VERBOSE: Importing cmdlet 'Add-VMNetworkAdapterRoutingDomainMapping'.
VERBOSE: Importing cmdlet 'Add-VMRemoteFx3dVideoAdapter'.
VERBOSE: Importing cmdlet 'Add-VMScsiController'.
VERBOSE: Importing cmdlet 'Add-VMStoragePath'.
VERBOSE: Importing cmdlet 'Add-VMSwitch'.
VERBOSE: Importing cmdlet 'Add-VMSwitchExtensionPortFeature'.
VERBOSE: Importing cmdlet 'Add-VMSwitchExtensionSwitchFeature'.
VERBOSE: Importing cmdlet 'Add-VMSwitchTeamMember'.
VERBOSE: Importing cmdlet 'Checkpoint-VM'.
VERBOSE: Importing cmdlet 'Compare-VM'.
VERBOSE: Importing cmdlet 'Complete-VMFailover'.
VERBOSE: Importing cmdlet 'Connect-VMNetworkAdapter'.
VERBOSE: Importing cmdlet 'Connect-VMSan'.
VERBOSE: Importing cmdlet 'Convert-VHD'.
VERBOSE: Importing cmdlet 'Copy-VMFile'.
VERBOSE: Importing cmdlet 'Debug-VM'.
VERBOSE: Importing cmdlet 'Disable-VMConsoleSupport'.
VERBOSE: Importing cmdlet 'Disable-VMEventing'.
VERBOSE: Importing cmdlet 'Disable-VMIntegrationService'.
VERBOSE: Importing cmdlet 'Disable-VMMigration'.
VERBOSE: Importing cmdlet 'Disable-VMRemoteFXPhysicalVideoAdapter'.
VERBOSE: Importing cmdlet 'Disable-VMResourceMetering'.
VERBOSE: Importing cmdlet 'Disable-VMSwitchExtension'.
VERBOSE: Importing cmdlet 'Disable-VMTPM'.
VERBOSE: Importing cmdlet 'Disconnect-VMNetworkAdapter'.
VERBOSE: Importing cmdlet 'Disconnect-VMSan'.
VERBOSE: Importing cmdlet 'Dismount-VHD'.
VERBOSE: Importing cmdlet 'Dismount-VMHostAssignableDevice'.
VERBOSE: Importing cmdlet 'Enable-VMConsoleSupport'.
VERBOSE: Importing cmdlet 'Enable-VMEventing'.
VERBOSE: Importing cmdlet 'Enable-VMIntegrationService'.
VERBOSE: Importing cmdlet 'Enable-VMMigration'.
VERBOSE: Importing cmdlet 'Enable-VMRemoteFXPhysicalVideoAdapter'.
VERBOSE: Importing cmdlet 'Enable-VMReplication'.
VERBOSE: Importing cmdlet 'Enable-VMResourceMetering'.
VERBOSE: Importing cmdlet 'Enable-VMSwitchExtension'.
VERBOSE: Importing cmdlet 'Enable-VMTPM'.
VERBOSE: Importing cmdlet 'Export-VM'.
VERBOSE: Importing cmdlet 'Export-VMSnapshot'.
VERBOSE: Importing cmdlet 'Get-VHD'.
VERBOSE: Importing cmdlet 'Get-VHDSet'.
VERBOSE: Importing cmdlet 'Get-VHDSnapshot'.
VERBOSE: Importing cmdlet 'Get-VM'.
VERBOSE: Importing cmdlet 'Get-VMAssignableDevice'.
VERBOSE: Importing cmdlet 'Get-VMBios'.
VERBOSE: Importing cmdlet 'Get-VMComPort'.
VERBOSE: Importing cmdlet 'Get-VMConnectAccess'.
VERBOSE: Importing cmdlet 'Get-VMDvdDrive'.
VERBOSE: Importing cmdlet 'Get-VMFibreChannelHba'.
VERBOSE: Importing cmdlet 'Get-VMFirmware'.
VERBOSE: Importing cmdlet 'Get-VMFloppyDiskDrive'.
VERBOSE: Importing cmdlet 'Get-VMGpuPartitionAdapter'.
VERBOSE: Importing cmdlet 'Get-VMGroup'.
VERBOSE: Importing cmdlet 'Get-VMHardDiskDrive'.
VERBOSE: Importing cmdlet 'Get-VMHost'.
VERBOSE: Importing cmdlet 'Get-VMHostAssignableDevice'.
VERBOSE: Importing cmdlet 'Get-VMHostCluster'.
VERBOSE: Importing cmdlet 'Get-VMHostNumaNode'.
VERBOSE: Importing cmdlet 'Get-VMHostNumaNodeStatus'.
VERBOSE: Importing cmdlet 'Get-VMHostSupportedVersion'.
VERBOSE: Importing cmdlet 'Get-VMIdeController'.
VERBOSE: Importing cmdlet 'Get-VMIntegrationService'.
VERBOSE: Importing cmdlet 'Get-VMKeyProtector'.
VERBOSE: Importing cmdlet 'Get-VMKeyStorageDrive'.
VERBOSE: Importing cmdlet 'Get-VMMemory'.
VERBOSE: Importing cmdlet 'Get-VMMigrationNetwork'.
VERBOSE: Importing cmdlet 'Get-VMNetworkAdapter'.
VERBOSE: Importing cmdlet 'Get-VMNetworkAdapterAcl'.
VERBOSE: Importing cmdlet 'Get-VMNetworkAdapterExtendedAcl'.
VERBOSE: Importing cmdlet 'Get-VMNetworkAdapterFailoverConfiguration'.
VERBOSE: Importing cmdlet 'Get-VMNetworkAdapterIsolation'.
VERBOSE: Importing cmdlet 'Get-VMNetworkAdapterRoutingDomainMapping'.
VERBOSE: Importing cmdlet 'Get-VMNetworkAdapterTeamMapping'.
VERBOSE: Importing cmdlet 'Get-VMNetworkAdapterVlan'.
VERBOSE: Importing cmdlet 'Get-VMPartitionableGpu'.
VERBOSE: Importing cmdlet 'Get-VMProcessor'.
VERBOSE: Importing cmdlet 'Get-VMRemoteFx3dVideoAdapter'.
VERBOSE: Importing cmdlet 'Get-VMRemoteFXPhysicalVideoAdapter'.
VERBOSE: Importing cmdlet 'Get-VMReplication'.
VERBOSE: Importing cmdlet 'Get-VMReplicationAuthorizationEntry'.
VERBOSE: Importing cmdlet 'Get-VMReplicationServer'.
VERBOSE: Importing cmdlet 'Get-VMResourcePool'.
VERBOSE: Importing cmdlet 'Get-VMSan'.
VERBOSE: Importing cmdlet 'Get-VMScsiController'.
VERBOSE: Importing cmdlet 'Get-VMSecurity'.
VERBOSE: Importing cmdlet 'Get-VMSnapshot'.
VERBOSE: Importing cmdlet 'Get-VMStoragePath'.
VERBOSE: Importing cmdlet 'Get-VMSwitch'.
VERBOSE: Importing cmdlet 'Get-VMSwitchExtension'.
VERBOSE: Importing cmdlet 'Get-VMSwitchExtensionPortData'.
VERBOSE: Importing cmdlet 'Get-VMSwitchExtensionPortFeature'.
VERBOSE: Importing cmdlet 'Get-VMSwitchExtensionSwitchData'.
VERBOSE: Importing cmdlet 'Get-VMSwitchExtensionSwitchFeature'.
VERBOSE: Importing cmdlet 'Get-VMSwitchTeam'.
VERBOSE: Importing cmdlet 'Get-VMSystemSwitchExtension'.
VERBOSE: Importing cmdlet 'Get-VMSystemSwitchExtensionPortFeature'.
VERBOSE: Importing cmdlet 'Get-VMSystemSwitchExtensionSwitchFeature'.
VERBOSE: Importing cmdlet 'Get-VMVideo'.
VERBOSE: Importing cmdlet 'Grant-VMConnectAccess'.
VERBOSE: Importing cmdlet 'Import-VM'.
VERBOSE: Importing cmdlet 'Import-VMInitialReplication'.
VERBOSE: Importing cmdlet 'Measure-VM'.
VERBOSE: Importing cmdlet 'Measure-VMReplication'.
VERBOSE: Importing cmdlet 'Measure-VMResourcePool'.
VERBOSE: Importing cmdlet 'Merge-VHD'.
VERBOSE: Importing cmdlet 'Mount-VHD'.
VERBOSE: Importing cmdlet 'Mount-VMHostAssignableDevice'.
VERBOSE: Importing cmdlet 'Move-VM'.
VERBOSE: Importing cmdlet 'Move-VMStorage'.
VERBOSE: Importing cmdlet 'New-VFD'.
VERBOSE: Importing cmdlet 'New-VHD'.
VERBOSE: Importing cmdlet 'New-VM'.
VERBOSE: Importing cmdlet 'New-VMGroup'.
VERBOSE: Importing cmdlet 'New-VMReplicationAuthorizationEntry'.
VERBOSE: Importing cmdlet 'New-VMResourcePool'.
VERBOSE: Importing cmdlet 'New-VMSan'.
VERBOSE: Importing cmdlet 'New-VMSwitch'.
VERBOSE: Importing cmdlet 'Optimize-VHD'.
VERBOSE: Importing cmdlet 'Optimize-VHDSet'.
VERBOSE: Importing cmdlet 'Remove-VHDSnapshot'.
VERBOSE: Importing cmdlet 'Remove-VM'.
VERBOSE: Importing cmdlet 'Remove-VMAssignableDevice'.
VERBOSE: Importing cmdlet 'Remove-VMDvdDrive'.
VERBOSE: Importing cmdlet 'Remove-VMFibreChannelHba'.
VERBOSE: Importing cmdlet 'Remove-VMGpuPartitionAdapter'.
VERBOSE: Importing cmdlet 'Remove-VMGroup'.
VERBOSE: Importing cmdlet 'Remove-VMGroupMember'.
VERBOSE: Importing cmdlet 'Remove-VMHardDiskDrive'.
VERBOSE: Importing cmdlet 'Remove-VMHostAssignableDevice'.
VERBOSE: Importing cmdlet 'Remove-VMKeyStorageDrive'.
VERBOSE: Importing cmdlet 'Remove-VMMigrationNetwork'.
VERBOSE: Importing cmdlet 'Remove-VMNetworkAdapter'.
VERBOSE: Importing cmdlet 'Remove-VMNetworkAdapterAcl'.
VERBOSE: Importing cmdlet 'Remove-VMNetworkAdapterExtendedAcl'.
VERBOSE: Importing cmdlet 'Remove-VMNetworkAdapterRoutingDomainMapping'.
VERBOSE: Importing cmdlet 'Remove-VMNetworkAdapterTeamMapping'.
VERBOSE: Importing cmdlet 'Remove-VMRemoteFx3dVideoAdapter'.
VERBOSE: Importing cmdlet 'Remove-VMReplication'.
VERBOSE: Importing cmdlet 'Remove-VMReplicationAuthorizationEntry'.
VERBOSE: Importing cmdlet 'Remove-VMResourcePool'.
VERBOSE: Importing cmdlet 'Remove-VMSan'.
VERBOSE: Importing cmdlet 'Remove-VMSavedState'.
VERBOSE: Importing cmdlet 'Remove-VMScsiController'.
VERBOSE: Importing cmdlet 'Remove-VMSnapshot'.
VERBOSE: Importing cmdlet 'Remove-VMStoragePath'.
VERBOSE: Importing cmdlet 'Remove-VMSwitch'.
VERBOSE: Importing cmdlet 'Remove-VMSwitchExtensionPortFeature'.
VERBOSE: Importing cmdlet 'Remove-VMSwitchExtensionSwitchFeature'.
VERBOSE: Importing cmdlet 'Remove-VMSwitchTeamMember'.
VERBOSE: Importing cmdlet 'Rename-VM'.
VERBOSE: Importing cmdlet 'Rename-VMGroup'.
VERBOSE: Importing cmdlet 'Rename-VMNetworkAdapter'.
VERBOSE: Importing cmdlet 'Rename-VMResourcePool'.
VERBOSE: Importing cmdlet 'Rename-VMSan'.
VERBOSE: Importing cmdlet 'Rename-VMSnapshot'.
VERBOSE: Importing cmdlet 'Rename-VMSwitch'.
VERBOSE: Importing cmdlet 'Repair-VM'.
VERBOSE: Importing cmdlet 'Reset-VMReplicationStatistics'.
VERBOSE: Importing cmdlet 'Reset-VMResourceMetering'.
VERBOSE: Importing cmdlet 'Resize-VHD'.
VERBOSE: Importing cmdlet 'Restart-VM'.
VERBOSE: Importing cmdlet 'Restore-VMSnapshot'.
VERBOSE: Importing cmdlet 'Resume-VM'.
VERBOSE: Importing cmdlet 'Resume-VMReplication'.
VERBOSE: Importing cmdlet 'Revoke-VMConnectAccess'.
VERBOSE: Importing cmdlet 'Save-VM'.
VERBOSE: Importing cmdlet 'Set-VHD'.
VERBOSE: Importing cmdlet 'Set-VM'.
VERBOSE: Importing cmdlet 'Set-VMBios'.
VERBOSE: Importing cmdlet 'Set-VMComPort'.
VERBOSE: Importing cmdlet 'Set-VMDvdDrive'.
VERBOSE: Importing cmdlet 'Set-VMFibreChannelHba'.
VERBOSE: Importing cmdlet 'Set-VMFirmware'.
VERBOSE: Importing cmdlet 'Set-VMFloppyDiskDrive'.
VERBOSE: Importing cmdlet 'Set-VMGpuPartitionAdapter'.
VERBOSE: Importing cmdlet 'Set-VMHardDiskDrive'.
VERBOSE: Importing cmdlet 'Set-VMHost'.
VERBOSE: Importing cmdlet 'Set-VMHostCluster'.
VERBOSE: Importing cmdlet 'Set-VMKeyProtector'.
VERBOSE: Importing cmdlet 'Set-VMKeyStorageDrive'.
VERBOSE: Importing cmdlet 'Set-VMMemory'.
VERBOSE: Importing cmdlet 'Set-VMMigrationNetwork'.
VERBOSE: Importing cmdlet 'Set-VMNetworkAdapter'.
VERBOSE: Importing cmdlet 'Set-VMNetworkAdapterFailoverConfiguration'.
VERBOSE: Importing cmdlet 'Set-VMNetworkAdapterIsolation'.
VERBOSE: Importing cmdlet 'Set-VMNetworkAdapterRdma'.
VERBOSE: Importing cmdlet 'Set-VMNetworkAdapterRoutingDomainMapping'.
VERBOSE: Importing cmdlet 'Set-VMNetworkAdapterTeamMapping'.
VERBOSE: Importing cmdlet 'Set-VMNetworkAdapterVlan'.
VERBOSE: Importing cmdlet 'Set-VMPartitionableGpu'.
VERBOSE: Importing cmdlet 'Set-VMProcessor'.
VERBOSE: Importing cmdlet 'Set-VMRemoteFx3dVideoAdapter'.
VERBOSE: Importing cmdlet 'Set-VMReplication'.
VERBOSE: Importing cmdlet 'Set-VMReplicationAuthorizationEntry'.
VERBOSE: Importing cmdlet 'Set-VMReplicationServer'.
VERBOSE: Importing cmdlet 'Set-VMResourcePool'.
VERBOSE: Importing cmdlet 'Set-VMSan'.
VERBOSE: Importing cmdlet 'Set-VMSecurity'.
VERBOSE: Importing cmdlet 'Set-VMSecurityPolicy'.
VERBOSE: Importing cmdlet 'Set-VMSwitch'.
VERBOSE: Importing cmdlet 'Set-VMSwitchExtensionPortFeature'.
VERBOSE: Importing cmdlet 'Set-VMSwitchExtensionSwitchFeature'.
VERBOSE: Importing cmdlet 'Set-VMSwitchTeam'.
VERBOSE: Importing cmdlet 'Set-VMVideo'.
VERBOSE: Importing cmdlet 'Start-VM'.
VERBOSE: Importing cmdlet 'Start-VMFailover'.
VERBOSE: Importing cmdlet 'Start-VMInitialReplication'.
VERBOSE: Importing cmdlet 'Start-VMTrace'.
VERBOSE: Importing cmdlet 'Stop-VM'.
VERBOSE: Importing cmdlet 'Stop-VMFailover'.
VERBOSE: Importing cmdlet 'Stop-VMInitialReplication'.
VERBOSE: Importing cmdlet 'Stop-VMReplication'.
VERBOSE: Importing cmdlet 'Stop-VMTrace'.
VERBOSE: Importing cmdlet 'Suspend-VM'.
VERBOSE: Importing cmdlet 'Suspend-VMReplication'.
VERBOSE: Importing cmdlet 'Test-VHD'.
VERBOSE: Importing cmdlet 'Test-VMNetworkAdapter'.
VERBOSE: Importing cmdlet 'Test-VMReplicationConnection'.
VERBOSE: Importing cmdlet 'Update-VMVersion'.
VERBOSE: Importing cmdlet 'Wait-VM'.
VERBOSE: Importing function 'Disable-NetAdapter'.
VERBOSE: Importing function 'Disable-NetAdapterBinding'.
VERBOSE: Importing function 'Disable-NetAdapterChecksumOffload'.
VERBOSE: Importing function 'Disable-NetAdapterEncapsulatedPacketTaskOffload'.
VERBOSE: Importing function 'Disable-NetAdapterIPsecOffload'.
VERBOSE: Importing function 'Disable-NetAdapterLso'.
VERBOSE: Importing function 'Disable-NetAdapterPacketDirect'.
VERBOSE: Importing function 'Disable-NetAdapterPowerManagement'.
VERBOSE: Importing function 'Disable-NetAdapterQos'.
VERBOSE: Importing function 'Disable-NetAdapterRdma'.
VERBOSE: Importing function 'Disable-NetAdapterRsc'.
VERBOSE: Importing function 'Disable-NetAdapterRss'.
VERBOSE: Importing function 'Disable-NetAdapterSriov'.
VERBOSE: Importing function 'Disable-NetAdapterVmq'.
VERBOSE: Importing function 'Enable-NetAdapter'.
VERBOSE: Importing function 'Enable-NetAdapterBinding'.
VERBOSE: Importing function 'Enable-NetAdapterChecksumOffload'.
VERBOSE: Importing function 'Enable-NetAdapterEncapsulatedPacketTaskOffload'.
VERBOSE: Importing function 'Enable-NetAdapterIPsecOffload'.
VERBOSE: Importing function 'Enable-NetAdapterLso'.
VERBOSE: Importing function 'Enable-NetAdapterPacketDirect'.
VERBOSE: Importing function 'Enable-NetAdapterPowerManagement'.
VERBOSE: Importing function 'Enable-NetAdapterQos'.
VERBOSE: Importing function 'Enable-NetAdapterRdma'.
VERBOSE: Importing function 'Enable-NetAdapterRsc'.
VERBOSE: Importing function 'Enable-NetAdapterRss'.
VERBOSE: Importing function 'Enable-NetAdapterSriov'.
VERBOSE: Importing function 'Enable-NetAdapterVmq'.
VERBOSE: Importing function 'Get-NetAdapter'.
VERBOSE: Importing function 'Get-NetAdapterAdvancedProperty'.
VERBOSE: Importing function 'Get-NetAdapterBinding'.
VERBOSE: Importing function 'Get-NetAdapterChecksumOffload'.
VERBOSE: Importing function 'Get-NetAdapterEncapsulatedPacketTaskOffload'.
VERBOSE: Importing function 'Get-NetAdapterHardwareInfo'.
VERBOSE: Importing function 'Get-NetAdapterIPsecOffload'.
VERBOSE: Importing function 'Get-NetAdapterLso'.
VERBOSE: Importing function 'Get-NetAdapterPacketDirect'.
VERBOSE: Importing function 'Get-NetAdapterPowerManagement'.
VERBOSE: Importing function 'Get-NetAdapterQos'.
VERBOSE: Importing function 'Get-NetAdapterRdma'.
VERBOSE: Importing function 'Get-NetAdapterRsc'.
VERBOSE: Importing function 'Get-NetAdapterRss'.
VERBOSE: Importing function 'Get-NetAdapterSriov'.
VERBOSE: Importing function 'Get-NetAdapterSriovVf'.
VERBOSE: Importing function 'Get-NetAdapterStatistics'.
VERBOSE: Importing function 'Get-NetAdapterVmq'.
VERBOSE: Importing function 'Get-NetAdapterVmqQueue'.
VERBOSE: Importing function 'Get-NetAdapterVPort'.
VERBOSE: Importing function 'New-NetAdapterAdvancedProperty'.
VERBOSE: Importing function 'Remove-NetAdapterAdvancedProperty'.
VERBOSE: Importing function 'Rename-NetAdapter'.
VERBOSE: Importing function 'Reset-NetAdapterAdvancedProperty'.
VERBOSE: Importing function 'Restart-NetAdapter'.
VERBOSE: Importing function 'Set-NetAdapter'.
VERBOSE: Importing function 'Set-NetAdapterAdvancedProperty'.
VERBOSE: Importing function 'Set-NetAdapterBinding'.
VERBOSE: Importing function 'Set-NetAdapterChecksumOffload'.
VERBOSE: Importing function 'Set-NetAdapterEncapsulatedPacketTaskOffload'.
VERBOSE: Importing function 'Set-NetAdapterIPsecOffload'.
VERBOSE: Importing function 'Set-NetAdapterLso'.
VERBOSE: Importing function 'Set-NetAdapterPacketDirect'.
VERBOSE: Importing function 'Set-NetAdapterPowerManagement'.
VERBOSE: Importing function 'Set-NetAdapterQos'.
VERBOSE: Importing function 'Set-NetAdapterRdma'.
VERBOSE: Importing function 'Set-NetAdapterRsc'.
VERBOSE: Importing function 'Set-NetAdapterRss'.
VERBOSE: Importing function 'Set-NetAdapterSriov'.
VERBOSE: Importing function 'Set-NetAdapterVmq'.
VERBOSE: Importing function 'Find-NetRoute'.
VERBOSE: Importing function 'Get-NetCompartment'.
VERBOSE: Importing function 'Get-NetIPAddress'.
VERBOSE: Importing function 'Get-NetIPConfiguration'.
VERBOSE: Importing function 'Get-NetIPInterface'.
VERBOSE: Importing function 'Get-NetIPv4Protocol'.
VERBOSE: Importing function 'Get-NetIPv6Protocol'.
VERBOSE: Importing function 'Get-NetNeighbor'.
VERBOSE: Importing function 'Get-NetOffloadGlobalSetting'.
VERBOSE: Importing function 'Get-NetPrefixPolicy'.
VERBOSE: Importing function 'Get-NetRoute'.
VERBOSE: Importing function 'Get-NetTCPConnection'.
VERBOSE: Importing function 'Get-NetTCPSetting'.
VERBOSE: Importing function 'Get-NetTransportFilter'.
VERBOSE: Importing function 'Get-NetUDPEndpoint'.
VERBOSE: Importing function 'Get-NetUDPSetting'.
VERBOSE: Importing function 'New-NetIPAddress'.
VERBOSE: Importing function 'New-NetNeighbor'.
VERBOSE: Importing function 'New-NetRoute'.
VERBOSE: Importing function 'New-NetTransportFilter'.
VERBOSE: Importing function 'Remove-NetIPAddress'.
VERBOSE: Importing function 'Remove-NetNeighbor'.
VERBOSE: Importing function 'Remove-NetRoute'.
VERBOSE: Importing function 'Remove-NetTransportFilter'.
VERBOSE: Importing function 'Set-NetIPAddress'.
VERBOSE: Importing function 'Set-NetIPInterface'.
VERBOSE: Importing function 'Set-NetIPv4Protocol'.
VERBOSE: Importing function 'Set-NetIPv6Protocol'.
VERBOSE: Importing function 'Set-NetNeighbor'.
VERBOSE: Importing function 'Set-NetOffloadGlobalSetting'.
VERBOSE: Importing function 'Set-NetRoute'.
VERBOSE: Importing function 'Set-NetTCPSetting'.
VERBOSE: Importing function 'Set-NetUDPSetting'.
VERBOSE: Importing function 'Test-NetConnection'.
VERBOSE: Importing alias 'gip'.
VERBOSE: Importing alias 'TNC'.
Module loaded at 14:16:33.526
Creating Switch: DockerNAT...
```

The cmdlet doesn't seem to be able to finish creating the NAT switch.

**Comment by rneugeba** [comment]
Thanks for testing to quick. Could you try creating a VMswitch manually, something like:
```
New-VMSwitch -Name FooBar -SwitchType Internal
```
from an elevated powershell prompt?

**Comment by mmisztal1980** [comment]
@rneugeba the script froze while trying to create the switch ... Reinstalling Hyper-V

**Comment by rneugeba** [comment]
@mmisztal1980 you mean even the plain cmdlet `New-VMSwitch -Name FooBar -SwitchType Internal` freezes? If so I'll ping some folks at MSFT

**Comment by mmisztal1980** [comment]
@rneugeba yes. it doesn't complete

**Comment by mmisztal1980** [comment]
@rneugeba this is what I'm basically getting:

![docker-v13-hyperv-issue](https://cloud.githubusercontent.com/assets/1357057/22945660/3d7d9ac0-f2f5-11e6-9974-db273df2520a.png)

**Comment by rneugeba** [comment]
@mmisztal1980 thanks. I've pinged some MSFT folks. I've seen `New-VMSwitch` now on quite a few of our users systems

**Comment by mmisztal1980** [comment]
@rneugeba if they'd like to do a screen sharing session, to examine my pc, I'm happy to help

**Comment by rneugeba** [comment]
@mmisztal1980 thanks for the offer...I may get back to you on it, But for now could you do:
```
New-VMSwitch -Name FooBarBaz -SwitchType Internal
```
and let it hang for a while. And then, in a different elevated powershell:
```
get-vmswitch
get-netnat
get-containernetwork 
ipconfig /all
```
Also, execute `wevtutil gl System` and then either attach or email me the file from `c:\Windows\system32\WinEvt\Logs\System.evtx`. My email is rolf<dot>neugebauer<at>docker<dot>com

many thanks for the help

**Comment by mmisztal1980** [comment]
@rneugeba Here you go:

![docker-v13-ps](https://cloud.githubusercontent.com/assets/1357057/22948616/7ff5f3a2-f2ff-11e6-8d20-e32a5ad4df9a.png)
[ipconfig.txt](https://github.com/docker/for-win/files/775259/ipconfig.txt)

While trying to run wevutil I'm getting:
```
C:\Windows\system32>wevutil gl System
'wevutil' is not recognized as an internal or external command,
operable program or batch file.
```

So I've exported the Windows Logs\System manually:
[System.zip](https://github.com/docker/for-win/files/775273/System.zip)

Please let me know if this is it.

**Comment by rneugeba** [comment]
Thanks a lot

**Comment by mmisztal1980** [comment]
@rneugeba 
Interestingly, I found this in my device manager:

![docker-v13-hyperv-issue2](https://cloud.githubusercontent.com/assets/1357057/22949638/3e9b2054-f303-11e6-8e63-4867f2c1e8a0.png)

**Comment by dineshgovindasamy** [comment]
Those are leaked VNIC's.

Do you have any proprietary teaming solution or LBFO team created on this host?

**Comment by dineshgovindasamy** [comment]
@mmisztal1980 @rneugeba Could you please also run the following cleanupscript and share us the logs?

https://github.com/Microsoft/Virtualization-Documentation/tree/live/windows-server-container-tools/CleanupContainerHostNetworking

Please run it with CleanupContainerHostNetworking.ps1 with the -ForceCleanup parameter, this will perform log collection and also cleanup those leaked NICS.

**Comment by mmisztal1980** [comment]
@dineshgovindasamy 
I'm not sure what you mean? What is a LBFO or a teaming solution? You mean a VPN/SDN?

Logs attached. I've tried it twice, first using ```-Cleanup```, the using ```-ForceDeleteAllSwitches```

[PreCleanupState_HWOLYOCW.zip](https://github.com/docker/for-win/files/776305/PreCleanupState_HWOLYOCW.zip)
[PreCleanupState_PSVKYOAH.zip](https://github.com/docker/for-win/files/776306/PreCleanupState_PSVKYOAH.zip)
[PostCleanupState_PSVKYOAH.zip](https://github.com/docker/for-win/files/776307/PostCleanupState_PSVKYOAH.zip)

**Comment by rneugeba** [comment]
@JonCubed hit the same issue over at https://github.com/docker/for-win/issues/417#issuecomment-279139804. The diagnostics ID is 
41560D0B-E5AF-41B4-94BD-0C90C899004C/2017-02-11_22-59-1

**Comment by mmisztal1980** [comment]
@dineshgovindasamy 
For clarity, I'm using [zerotier](https://www.zerotier.com/) SDN/VPN to stay connected to one of my networks. The iface is visible in the device manager screenshot.

**Comment by mmisztal1980** [comment]
@dineshgovindasamy @rneugeba 
I've uninstalled zerotier and docker has started successfully.

**Comment by rneugeba** [comment]
@mmisztal1980 great to know. We recently had several issue with various AV/security softwares and looks like zerotier is another one to look out for

**Comment by rneugeba** [comment]
@yvesdelvaux do you have any AV/security  software installed? And if so could you try disabling?

**Comment by JonCubed** [comment]
@rneugeba @dineshgovindasamy let me know if you need me to do anything to help debug this issue, love to get docker working again

**Comment by rneugeba** [comment]
@JonCubed Have you tried what I suggested [here](https://github.com/docker/for-win/issues/482#issuecomment-279365617)?Most importantly the Hyper-V test?

Also do you run any third party AV/FW software?

**Comment by kevinykuo** [comment]
OK guys this is what I did to resolve the issue on my end. Go to "Turn Windows features on or off", disable all Hyper-V related features, reboot, then Docker should ask if it can enable and reboot for you. Let it do that and see if it's fixed. If not I'd probably try manually re-enabling Hyper-V. Good luck.

**Comment by JonCubed** [comment]
UI Test Diagnostic Id: 41560D0B-E5AF-41B4-94BD-0C90C899004C/2017-02-21_22-44-54

Hyper-V test fails on create with

```powershell
New-NetIPAddress : The object already exists.
At C:\Program Files\Docker\Docker\resources\MobyLinux.ps1:164 char:23
+ ... rkAdapter | New-NetIPAddress -AddressFamily IPv4 -IPAddress $switchAd ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (MSFT_NetIPAddress:ROOT/StandardCimv2/MSFT_NetIPAddress) [New-NetIPAddress
   ], CimException
    + FullyQualifiedErrorId : Windows System Error 5010,New-NetIPAddress
```

cc @rneugeba

**Comment by JonCubed** [comment]
@rneugeba just following up on this, anything else I can do to help with this?

**Comment by JonCubed** [comment]
I did some debugging turns out that I had a already had an assigned ip address for `10.0.75.1`.

```powershell
[DBG]: PS C:\WINDOWS\system32>> Get-NetIPAddress -IPAddress $switchAddress


IPAddress         : 10.0.75.1
InterfaceIndex    : 31
InterfaceAlias    : vEthernet (InternalNAT)
AddressFamily     : IPv4
Type              : Unicast
PrefixLength      : 24
PrefixOrigin      : Manual
SuffixOrigin      : Manual
AddressState      : Preferred
ValidLifetime     : Infinite ([TimeSpan]::MaxValue)
PreferredLifetime : Infinite ([TimeSpan]::MaxValue)
SkipAsSource      : False
PolicyStore       : ActiveStore

[DBG]: PS C:\Program Files\Docker\Docker\resources>> $networkAdapter | Format-List


Name                       : vEthernet (DockerNAT)
InterfaceDescription       : Hyper-V Virtual Ethernet Adapter #4
InterfaceIndex             : 112
MacAddress                 : 00-15-5D-4B-01-2E
MediaType                  : 802.3
PhysicalMediaType          : Unspecified
InterfaceOperationalStatus : Up
AdminStatus                : Up
LinkSpeed(Gbps)            : 10
MediaConnectionState       : Connected
ConnectorPresent           : False
DriverInformation          : Driver Date 2006-06-21 Version 10.0.14393.0 NDIS 6.50
```

this led me to `ln#157` in `MobyLinux.ps1`
```powershell
# Make sure the Switch has the right IP address
$networkAdapter = Get-NetAdapter | ? { $_.DeviceID -eq $vmNetAdapter.DeviceId }
if ($networkAdapter | Get-NetIPAddress -IPAddress $switchAddress -ea SilentlyContinue) {
        Write-Output "Using existing Switch IP address"
        return
}
```

if you remove `-ea SilentlyContinue` it throws the following error
```powershell
Get-NetIPAddress : No matching MSFT_NetIPAddress objects found by CIM query for instances of the 
ROOT/StandardCimv2/MSFT_NetIPAddress class on the  CIM server: SELECT * FROM MSFT_NetIPAddress  WHERE 
((IPAddress LIKE '10.0.75.1')) AND ((InterfaceIndex = 112)) AND ((InterfaceAlias LIKE 'vEthernet 
(DockerNAT)')). Verify query parameters and retry.
```
you can see it is also filtering by InterfaceAlias `vEthernet (DockerNAT)` but mine is called `vEthernet (InternalNAT)`

not piping `$networkAdapter` would make the if statement work, not sure that is what your really want though.

### To reproduce:
1. `.\MobyLixux.ps1 -Destroy`
2. `New-NetIPAddress -AddressFamily IPv4 -IPAddress 10.0.75.1 -PrefixLength (24) -InterfaceAlias "vEthernet (InternalNAT)" -ea Stop`
3. `.\MobyLinux.ps1 -Create`

### Workaround:
1. `.\MobyLixux.ps1 -Destroy`
2. `Remove-NetIPAddress -IPAddress 10.0.75.1`
3. `.\MobyLinux.ps1 -Create`
4. Restart docker for windows

cc @rneugeba

**Comment by megamindbrian** [comment]
This didn't work either.  Docker for Windows client is frozen for about 30 seconds then fails with same pipe not readable error.

**Comment by JonCubed** [comment]
@megamindbrian I had a problem with most recent version. I uninstalled it, removed all docker folders and server and virtual switch from hyper-v and then reinstalled and it worked. If you haven't tried this, give it a shot

**Comment by vvasilenok** [comment]
I have the same problem.
Diagnostic Id: CAB78893-6FDB-432D-B75F-E932445907AF/2017-07-20_10-15-51

[NamedPipeServer][Error  ] Pipe failure: Pipe is broken.    at System.IO.Pipes.PipeStream.WinIOError(Int32 errorCode)
   at System.IO.Pipes.PipeStream.BeginWriteCore(Byte[] buffer, Int32 offset, Int32 count, AsyncCallback callback, Object state)
   at System.IO.Stream.<>c.<BeginEndWriteAsync>b__53_0(Stream stream, ReadWriteParameters args, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncTrim[TInstance,TArgs](TInstance thisRef, TArgs args, Func`5 beginMethod, Func`3 endMethod)
   at System.IO.Stream.BeginEndWriteAsync(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.Stream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at System.IO.Stream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count)
   at Docker.Core.Pipe.NamedPipeServer.<HandleRequestAsync>d__12.MoveNext() in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 117

**Comment by tneward** [comment]
Just to add data points--this is happening on my Win10 install as well just recently. Going to start trying some of the steps presented here, but if there's data I can provide to help track this down, lmk.

**Comment by megamindbrian** [comment]
I uninstalled and reinstalled edge and it worked.  Came up again later
after an upgrade and uninstalled and reinstalled and it worked.

**Comment by tneward** [comment]
Yeah, I tried that yesterday but the hanging behavior just started, so I'm thinking maybe after an uninstall things got left in a weird state. That's why I'm going to try some of the options mentioned in here as well as doing an uninstall/reinstall.

**Comment by rn** [comment]
closing as the OP is pretty old
